L/libfabric/build_tarballs.jl

```diff
@@ -0,0 +1,34 @@
+using BinaryBuilder, Pkg
+
+name = "libfabric"
+version = v"2.0.0"
+
+sources = [
+    GitSource("https://github.com/ofiwg/libfabric", "2ee68f6051e90a59d7550d94a331fdf5e038db90"),
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/libfabric
+
+./autogen.sh
+./configure --build=${MACHTYPE} --host=${target} --prefix=${prefix} --with-dlopen
+make -j${nproc}
+make install
+"""
+
+# libfabric only builds on Linux, OS X, and FreeBSD
+platforms = supported_platforms(; exclude=Sys.iswindows)
+
+products = [
+    LibraryProduct("libfabric", :libfabric),
+]
+
+dependencies = [
+    # We need the header file `bits/types/struct_iovec.h` that was added in glibc 2.26
+    BuildDependency(PackageSpec(name="Glibc_jll", version=v"2.26"); platforms=glibc_platforms))
```

Comment:
If you're going to use this in any of the MPI libraries, that'd prevent all packages using MPI to work on the clusters of my university, which I'm not particularly happy about.

Reply:
The package builds on all platforms except `x86_64` where it requires glibc 2.26. I guess only `x86_64` uses old glibcs.

This is just a build dependency to find a particular header file. Would that still be a disaster?

Reply:
I switched to an older version. No `Glibc_jll` depency any more.

Reply:
> This is just a build dependency to find a particular header file. Would that still be a disaster?

Yes, if final library ends up using glibc > 2.17 symbols anyway (which may or may not the case, depending on what C stdlib functions it needs)